patch/cisco-acpi-spi-nor.patch

```diff
@@ -0,0 +1,43 @@
+From a08c84878b4e26e04015737b5e85b0d90f8c1b51 Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Tue, 8 Jun 2021 10:30:42 -0700
+Subject: [PATCH] mtd: m25p80 spi driver update to support ACPI table match
+
+Current m25p80 spi nor driver does not support ACPI table match.
+Configuration done through ACPI tables is not recognized by driver.
+Cisco 8000 platform cofigures NOR flash partition information
+through APCI table which is not recognized by m25p80.
+
+Added support in m25p80 to perform ACPI table match.
+
```

Comment:
Can you please provide the ASL code (fetch with `acpidump`, `acpixtract`, and `iasl -d`)? Please mention what firmware is used on the Cisco device.

Reply:
Please see ASL version

cisco@sonic:~$ iasl -d
Intel ACPI Component Architecture
ASL+ Optimizing Compiler/Disassembler version 20181213
Copyright (c) 2000 - 2018 Intel Corporation

ACPI dump
cisco@sonic:~$ sudo acpidump > acpi.dat
cisco@sonic:~$ ls
acpi.dat
cisco@sonic:~$ acpixtract -a acpi.dat 

Intel ACPI Component Architecture
ACPI Binary Table Extraction Utility version 20181213
Copyright (c) 2000 - 2018 Intel Corporation

  SSDT -  535404 bytes written (0x00082B6C) - ssdt1.dat
  MCFG -      60 bytes written (0x0000003C) - mcfg.dat
  EINJ -     336 bytes written (0x00000150) - einj.dat
  APIC -     230 bytes written (0x000000E6) - apic.dat
  SSDT -  933078 bytes written (0x000E3CD6) - ssdt2.dat
  TPM2 -      52 bytes written (0x00000034) - tpm2.dat
  BOOT -      40 bytes written (0x00000028) - boot.dat
  SLIT -     108 bytes written (0x0000006C) - slit.dat
  ERST -     560 bytes written (0x00000230) - erst.dat
  DSDT -   89633 bytes written (0x00015E21) - dsdt.dat
  SRAT -     216 bytes written (0x000000D8) - srat.dat
  UEFI -     566 bytes written (0x00000236) - uefi1.dat
  WDAT -     404 bytes written (0x00000194) - wdat.dat
  HEST -     168 bytes written (0x000000A8) - hest.dat
  BERT -      48 bytes written (0x00000030) - bert.dat
  SSDT -     758 bytes written (0x000002F6) - ssdt3.dat
  DMAR -     212 bytes written (0x000000D4) - dmar.dat
  FACP -     268 bytes written (0x0000010C) - facp.dat
  FPDT -      68 bytes written (0x00000044) - fpdt.dat
  SSDT -    1364 bytes written (0x00000554) - ssdt4.dat
  PRAD -     209 bytes written (0x000000D1) - prad.dat
  DBGP -      52 bytes written (0x00000034) - dbgp.dat
  HPET -      56 bytes written (0x00000038) - hpet.dat
  UEFI -      66 bytes written (0x00000042) - uefi2.dat
  FACS -      64 bytes written (0x00000040) - facs.dat
  BGRT -      56 bytes written (0x00000038) - bgrt.dat

cisco@sonic:~$ iasl -d ssdt1.dat 

Intel ACPI Component Architecture
ASL+ Optimizing Compiler/Disassembler version 20181213
Copyright (c) 2000 - 2018 Intel Corporation

Input file ssdt1.dat, Length 0x82B6C (535404) bytes
ACPI: SSDT 0x0000000000000000 082B6C (v02 CISCO  8800_RP  00000001 INTL 20181213)
Pass 1 parse of [SSDT]
Pass 2 parse of [SSDT]
Parsing Deferred Opcodes (Methods/Buffers/Packages/Regions)

Parsing completed
Disassembly completed
ASL Output:    ssdt1.dsl - 3191246 bytes

ASL code for the MTD partition

Please find attached ssdt1.dsl file which has ASL code for MTD partition
[ssdt1.zip](https://github.com/Azure/sonic-linux-kernel/files/6640629/ssdt1.zip)

Reply:
Thank you for your comment. Please edit your comment to mark up the paste as code block, indenting them with four spaces or tab or enclosing the block with ```, so it’s better readable.